manifest/caches/postgres.py

```diff
@@ -0,0 +1,110 @@
+"""SQLite cache."""
+import logging
+from typing import Any, Dict, Union
+
+import pg8000
+import sqlalchemy
+from sqlalchemy.dialects.postgresql import HSTORE
+from sqlalchemy import Column, Integer, String, Boolean, DateTime
+
+
+from manifest.caches.cache import Cache
+
+logging.getLogger("sqlitedict").setLevel(logging.WARNING)
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.pool import NullPool
+
+Base = declarative_base()
+
+
+class Request(Base):
+    __tablename__ = "requests"
+    key = Column(String, primary_key=True)
+    response = Column(String)   # FIXME: this should be an hstore ideally, but I don't want to set it up on GCP
+
+
+class PostgreSQLCache(Cache):
+    """A PostgreSQL cache for request/response pairs."""
+
+    def connect(
+        self, 
+        connection_str: str, 
+        cache_args: Dict[str, Any]
+    ) -> None:
+        """
+        Connect to client.
+
+        Args:
+            connection_str: connection string.
+            cache_args: arguments for cache.
+        """
+        from google.cloud.sql.connector import Connector
+        connector = Connector()
+
+        def getconn():
+            conn = connector.connect(
+                connection_str,
+                "pg8000",
+                user=cache_args.pop("user"),
```

Comment:
We probably want to set these as `pg_user` or `cache_user` or something so it's clear this is a cache param. Ideally we should refactor to get a clearer set of input params for a cache like we have for request, but this is fine for now. It'll happen once there are 2-3 caches that need params.